lib/compress.js

```diff
@@ -4655,4 +4655,36 @@ merge(Compressor.prototype, {
     OPT(AST_PrefixedTemplateString, function(self, compressor){
         return self;
     });
+
+    OPT(AST_ObjectKeyVal, function(self, compressor){
+        // p:function(){} ---> p(){}
+        // p:function*(){} ---> *p(){}
+        // p:async function(){} ---> async p(){}
+        // p:()=>{} ---> p(){}
```

Comment:
Thanks - coded that up before seeing your comment. See test: [`prop_arrow_with_this`](https://github.com/mishoo/UglifyJS2/pull/2199/files#diff-07c99e5886bd2b39a05451db475c6896R654).